packages/web/src/components/tile/Tile.tsx

```diff
@@ -0,0 +1,38 @@
+import { ReactNode, ComponentType, ComponentProps, ElementType } from 'react'
+
+import cn from 'classnames'
+
+import styles from './Tile.module.css'
+
+type TileOwnProps<TileComponentType extends ElementType = 'div'> = {
+  children: ReactNode
+  size?: 'small' | 'medium' | 'large'
+  as?: TileComponentType
+}
+
+export type TileProps<TileComponentType extends ElementType> =
+  TileOwnProps<TileComponentType> &
+    Omit<ComponentProps<TileComponentType>, keyof TileOwnProps>
+
+export const Tile = <
+  T extends ElementType = ComponentType<ComponentProps<'div'>>
+>(
+  props: TileProps<T>
+) => {
+  const {
+    children,
+    size,
```

Comment:
should size have a default value?

Reply:
so i was thinking not providing it allows the content to fill however it wants for added flexibility for now.

Reply:
cool idea, im into it